lib/internal.js

```diff
@@ -31,7 +31,7 @@ let clientObject;
         if(!path){
             throw new Error('Path is needed to send request.');
         }
-        requestOptions.path = mlutil.databaseParam(clientObject.getConnectionParams(), path, '?')
+        requestOptions.path = mlutil.newRequestOptions(clientObject.getConnectionParams(), path, '?').path;
```

Comment:
This looks like it can be simplified by changing the `initialize` method (which is misspelled, it's missing an "i") to not instantiate the `requestOptions` object. So remove line 26 and then change this to :

    requestOptions = mlutil.newRequestOptions(connection params, path , '?'); 

That also allows for line 21 to be removed - I don't think there's any benefit to keeping `requestOptions` around as state, it's simpler to create it each time that `sendRequest` is invoked.

Reply:
This ticket was for refactoring "databaseParam" not internal.js. This file has few problems while refactoring but I did not want to change lot of Phil's code. We can open a new Jira ticket.

Reply:
We'll use a separate ticket to address `let requestOptions`.